js/components/channel_messages.jsx

```diff
@@ -36,6 +36,7 @@ var ChannelMessages = React.createClass({
   },
 
   componentWillUnmount: function() {
+    MessageStore.removeListener('change', this._messageStoreChange);
```

Comment:
Browse to About then back to Chat and check out the console. If you log
MessageStore.all() you'll see dups.

On Fri, Apr 3, 2015 at 6:33 PM, Jack Noble <notifications@github.com> wrote:

> In js/components/channel_messages.jsx
> <https://github.com/rorysaur/quack/pull/60#discussion_r27765333>:
>
> > @@ -36,6 +36,7 @@ var ChannelMessages = React.createClass({
> >    },
> >
> >    componentWillUnmount: function() {
> > +    MessageStore.removeListener('change', this._messageStoreChange);
>
> Ah, thats cool that React throws an error when that happens. Should be
> done anyway. Do you know why there is a 'two children with the same key
> warning'?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rorysaur/quack/pull/60/files#r27765333>.
>